### DISCUSSION TITLE
what does the "doc" mean in the project's title

### DISCUSSION THREAD

**Discussion by asmwarrior** [onqtam/doctest#777] [General]
Does it mean "doctor"? Or other things?

Thanks.

**Comment by onqtam** [comment]
comes from documentation

**Reply by asmwarrior** [reply]
Sorry, I still not quite understand its meaning.
documentation?
doctest means: "documentation" test?

**Reply by Epixu** [reply]
Usually, one writes documentation as comments near the relevant source code itself. 
Analogous to that, doctest's approach allows for tests to be written near the relevant code that you're testing, as opposed to having a completely different dedicated files for it.

**Reply by onqtam** [reply]
yes, that's the point.

In python where doctest originally comes from, "doctests" were tests written as comments near the code, but for C++ I wanted to make writing actual code as tests easy near the code and used the same name.
https://en.wikipedia.org/wiki/Doctest

**Reply by onqtam** [reply]
actual compilable & up-to-date test code as embedded documentation

**Reply by asmwarrior** [reply]
Thanks for the replies, I understand now.
